really-core/src/main/scala/io/really/gorilla/ObjectSubscriber.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2014-2015 Really Inc. <http://really.io>
+ */
+
+package io.really.gorilla
+
+import akka.actor._
+import akka.util.Timeout
+import io.really.gorilla.GorillaEventCenter.ReplayerSubscribed
+import io.really.gorilla.SubscriptionManager.{ UpdateSubscriptionFields, Unsubscribe }
+import io.really.ReallyGlobals
+import io.really.model.persistent.ModelRegistry
+import io.really.model.persistent.ModelRegistry.CollectionActorMessage.GetModel
+import io.really.model.persistent.ModelRegistry.ModelResult
+import io.really.model.persistent.ModelRegistry.ModelResult.ModelFetchError
+import io.really.protocol.FieldUpdatedOp
+import io.really.protocol.ProtocolFormats.PushMessageWrites.{ Created, Updated, Deleted }
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import akka.pattern.{ AskTimeoutException, ask, pipe }
+import io.really.model.Model
+import io.really.protocol.SubscriptionFailure
+import io.really.protocol.SubscriptionFailure.SubscriptionFailureWrites
+
+class ObjectSubscriber(rSubscription: RSubscription, globals: ReallyGlobals) extends Actor with ActorLogging
+    with Stash {
+
+  val r = rSubscription.r
+  val logTag = s"ObjectSubscriber ${rSubscription.pushChannel.path}$$$r"
+
+  var fields = rSubscription.fields.getOrElse(Set.empty)
+
+  val shotgun = context.system.scheduler.scheduleOnce(globals.config.GorillaConfig.waitForReplayer, self, PoisonPill)
+
+  def subscriptionFailed(errorCode: Int, reason: String) = {
+    log.error(s"$logTag is going to die since the subscription failed because of: $reason\n error code: $errorCode")
+    rSubscription.pushChannel ! SubscriptionFailureWrites.writes(SubscriptionFailure(r, errorCode, "Internal Server Error"))
+    context.stop(self)
+  }
+
+  def commonHandler: Receive = {
+    case Unsubscribe =>
+      log.debug(s"$logTag is going to die since it got an unsubscribe request")
+      rSubscription.requestDelegate ! Unsubscribe
+      context.stop(self)
+    case SubscriptionFailure(r, errorCode, reason) =>
```

Comment:
I am sure of that, you can test it too, the replayer send this message here 
```
def shutdownWithObjectSubscriptionAck(reason: String) = {
    shutdown(reason)
    objectSubscriber ! SubscriptionFailure(r, 500, reason)
  }
```
What happens here is, first the replayer will terminate and the `objectSubscriber` will receive terminate message as it watch the replayer and will enter this case which will send the first failure message to the client
```
case Terminated(actor) =>
      subscriptionFailed(505, "Associated replayer stopped")
``` 
In the second line you send `objectSubscriber ! SubscriptionFailure(r, 500, reason)` which will enter this case 
```
case SubscriptionFailure(r, errorCode, reason) =>
      subscriptionFailed(errorCode, "Internal Server Error")
```
and the client will receive the second message